### PR TITLE
Generic Routes Validations #39524

### PR DESCRIPTION
**Pull request by shashankbaranwal**
- Introduction to Middleware.
- Validating all the Routes using Middleware
- Logging Request using morgan

### CODE REVIEW

**Review by namanpar — CHANGES_REQUESTED** [review]
Fix the lint

File: src/controllers/trainee/Controller.ts [hunk 1]
```diff
@@ -1,90 +1,91 @@
-import { Request, Response, NextFunction } from "express";
+import { Request, Response, NextFunction } from 'express';
+// tslint:disable-next-line: class-name
 class traineeController {
```
Comment (namanpar) [inline src/controllers/trainee/Controller.ts:3]: ClassName shoulde be proper

File: src/controllers/trainee/controller.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { Request, Response, NextFunction } from 'express';
+// tslint:disable-next-line: class-name
+class traineeController {
```
Comment (namanpar) [inline src/controllers/trainee/controller.ts:3]: ClassName should be proper

File: src/controllers/trainee/controller.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                }
+             });
+
+        } catch (err) {
+            console.log('inside err', err);
+        }
+
+    }
+
+    Delete( req: Request, res: Response, next: NextFunction ) {
```
Comment (namanpar) [inline src/controllers/trainee/controller.ts:75]: method name should be proper

File: src/libs/routes/errorHandler.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import { Request, Response, NextFunction } from 'express';
 
-export default function errorHandler (err, req: Request, res: Response, next: NextFunction ) {
+export default function errorHandler(err, req: Request, res: Response, next: NextFunction) {
```
Comment (namanpar) [inline src/libs/routes/errorHandler.ts:3]: Use es6

File: src/libs/routes/errorHandler.ts [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
-        message: err.message || "Error",
+        message: err.message || 'Error',
         timestamp: new Date()
-        
-    })
-}
\ No newline at end of file
+    });
+}
+
```
Comment (namanpar) [inline src/libs/routes/errorHandler.ts]: use destructing and handle undefined value cases

File: src/libs/routes/notFoundRoute.ts [hunk 6]
```diff
@@ -1,8 +1,8 @@
-import { Request, Response, NextFunction } from 'express'
+import { Request, Response, NextFunction } from 'express';
 
 export default function notFoundRoute(req: Request, res: Response, next: NextFunction) {
```
Comment (namanpar) [inline src/libs/routes/notFoundRoute.ts:3]: use es6

**Review by namanpar — CHANGES_REQUESTED** [review]
Please do the requested changes

File: src/controllers/trainee/Controller.ts (hunk 1 above)
Comment (shashankbaranwal) [reply src/controllers/trainee/Controller.ts]: used ES6 and proper naming

File: src/controllers/trainee/controller.ts (hunk 2 above)
Comment (shashankbaranwal) [reply src/controllers/trainee/controller.ts]: used ES6 and proper naming

File: src/controllers/trainee/controller.ts (hunk 3 above)
Comment (shashankbaranwal) [reply src/controllers/trainee/controller.ts]: used ES6 and proper naming method

File: src/libs/routes/errorHandler.ts (hunk 4 above)
Comment (shashankbaranwal) [reply src/libs/routes/errorHandler.ts]: Used ES6 please check it

File: src/libs/routes/errorHandler.ts (hunk 5 above)
Comment (shashankbaranwal) [reply src/libs/routes/errorHandler.ts]: Used destructing and handle all undefined value cases

File: src/libs/routes/notFoundRoute.ts (hunk 6 above)
Comment (shashankbaranwal) [reply src/libs/routes/notFoundRoute.ts]: Used ES6 please check it

File: src/controllers/trainee/controller.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                    name: 'Trainee',
+                    address: 'Noida'
+                }
+             });
+
+        } catch (err) {
+            console.log('inside err', err);
+
+        }
+
```
Comment (namanpar) [inline src/controllers/trainee/controller.ts:88]: try to handle error on input value

**Review by namanpar — CHANGES_REQUESTED** [review]

**Review by namanpar — APPROVED** [review]

**Pull request merged**